### PR TITLE
User Story 7571836: .Net SDK changes for supporting send by userId

### PR DESCRIPTION
**Pull request by olgaakv**
Updated Installation.cs with add UserId property.
Updated ApiVersion to 2020-06.
Updated Registrationdescription.cs to support userId as a valid tag similar to changes for RegistrationDescription class in backend.

### CODE REVIEW

File: src/Microsoft.Azure.NotificationHubs/Messaging/ManagementStrings.cs [hunk 1]
```diff
@@ -140,7 +140,7 @@ internal static class ManagementStrings
         public const string PnsErrorDetailsUri = "PnsErrorDetailsUri";
 
         public const string ApiVersionName = "api-version";
-        public const string ApiVersion = "2017-04";
+        public const string ApiVersion = "2020-06";
```
[the remark this answers is not in the document]
Comment (shantnugehlot) [reply src/Microsoft.Azure.NotificationHubs/Messaging/ManagementStrings.cs:143]: I think it should be part of 4.1.0

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/SendNotificationAsync_SendAppleNativeNotification_ByInstallationId_GetSuccessfulResultBack.http [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+   "HttpCalls":[
+      {
+         "RequestUriPath":"/sdk-sample-nh/messages?api-version=2020-06",
+         "StatusCode":201,
+         "Headers":{
+            "Strict-Transport-Security":"max-age=2592000",
+            "Date":"Thu, 23 Jul 2020 04:27:53 GMT",
+            "Location":"https://sdk-sample-namespace.notificationhub.int7.windows-int.net/sdk-sample-nh/messages/2e7f83cc61d44e3d9397c86939918508-20200723042753-586152-10?api-version=2020-06",
```
Comment (shantnugehlot) [inline src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/SendNotificationAsync_SendAppleNativeNotification_ByInstallationId_GetSuccessfulResultBack.http:9]: I think the sdk-sample-namespace is in prod not int

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithUserIdAndRemoveUserIdThroughPatch_GetInstallationWithoutUserIdBack.http [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            "Transfer-Encoding":"chunked",
+            "Strict-Transport-Security":"max-age=2592000"
+         },
+         "ContentHeaders":{
+            "Content-Type":"application/xml; charset=utf-8"
+         },
+         "Content":"{\"installationId\":\"d578eed1-1eef-4dfa-badf-a3a0c14e71e4\",\"pushChannel\":\"62BCD3484834B7E2DC56733AC5509259230CC1BA82BA1AE7D06E91908A05A8FA\",\"pushChannelExpired\":false,\"platform\":\"apns\",\"expirationTime\":\"9999-12-31T23:59:59.9999999Z\"}"
+      }
+   ],
+   "Guids":[
```
[the remark this answers is not in the document]
Comment (stankovski) [reply src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithUserIdAndRemoveUserIdThroughPatch_GetInstallationWithoutUserIdBack.http:73]: What do you mean?

File: src/Microsoft.Azure.NotificationHubs/Messaging/ManagementStrings.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (stankovski) [reply src/Microsoft.Azure.NotificationHubs/Messaging/ManagementStrings.cs:143]: Agree. I'll create a PR to update 4.0.0 branch with new API version.

**Comment by stankovski** [comment]
Please update the PR to target branch `version-4.1.0`.

File: src/Microsoft.Azure.NotificationHubs/Messaging/ManagementStrings.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (iTOYS) [reply src/Microsoft.Azure.NotificationHubs/Messaging/ManagementStrings.cs:143]: changed target

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/SendNotificationAsync_SendAppleNativeNotification_ByInstallationId_GetSuccessfulResultBack.http (hunk 2 above)
Comment (iTOYS) [reply src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/SendNotificationAsync_SendAppleNativeNotification_ByInstallationId_GetSuccessfulResultBack.http:9]: It's a mock data, we can write here whatever we want in proper format :)

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs [hunk 4]
```diff
@@ -40,8 +40,10 @@ namespace Microsoft.Azure.NotificationHubs
     public abstract class RegistrationDescription : EntityDescription
     {
         internal const string TemplateRegistrationType = "template";
-        internal static Regex SingleTagRegex = new Regex(@"^((\$InstallationId:\{[\w-_@#.:=]+\})|([\w-_@#.:]+))$", RegexOptions.IgnoreCase);
-        internal static Regex TagRegex = new Regex(@"^(((\$InstallationId:\{[\w-_@#.:=]+\})+?(,[\w-_@#.:]+)*)|(([\w-_@#.:]+)(,[\w-_@#.:]+)*((,\$InstallationId:\{[\w-_@#.:=]+\})?(,[\w-_@#.:]+)*)))$", RegexOptions.IgnoreCase);
+        internal static Regex SingleTagRegex = new Regex(@"^(\$(InstallationId|UserId):\{[-_@#.:=\w]+\}|[-_@#.:\w]+)$", RegexOptions.IgnoreCase);
+        internal static Regex UserIdRegex = new Regex(@"\$UserId:\{([-_@#.:=\w]+)\}", RegexOptions.IgnoreCase);
```
[the remark this answers is not in the document]
Comment (iTOYS) [reply src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs:44]: Where? as I can see Regex's the same

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithUserIdAndRemoveUserIdThroughPatch_GetInstallationWithoutUserIdBack.http (hunk 3 above)
[the remark this answers is not in the document]
Comment (iTOYS) [reply src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithUserIdAndRemoveUserIdThroughPatch_GetInstallationWithoutUserIdBack.http:73]: it's internal data for tests

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithoutUserIdAndAddUserIdThroughPatch_GetInstallationWithAddedUserIdBack.http [hunk 5]
```diff
@@ -0,0 +1,63 @@
+{
```
[the remark this answers is not in the document]
Comment (iTOYS) [reply src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithoutUserIdAndAddUserIdThroughPatch_GetInstallationWithAddedUserIdBack.http:1]: We use mock data as a response, this tests are like unit test

File: src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithUserIdAndRemoveUserIdThroughPatch_GetInstallationWithoutUserIdBack.http [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            "Server":"Microsoft-HTTPAPI/2.0",
+            "Strict-Transport-Security":"max-age=2592000"
+         },
+         "ContentHeaders":{
+            "Content-Length":"0"
+         },
+         "Content":null
+      },
+      {
+         "RequestUriPath":"/sdk-sample-nh/installations/d578eed1-1eef-4dfa-badf-a3a0c14e71e4?api-version=2020-06",
```
[the remark this answers is not in the document]
Comment (iTOYS) [reply src/Microsoft.Azure.NotificationHubs.DotNetCore.Tests/MockData/PatchInstallationAsync_CreateInstallationWithUserIdAndRemoveUserIdThroughPatch_GetInstallationWithoutUserIdBack.http:31]: It's a mock response

**Pull request merged**